internal/pkg/platform/platform_matcher.go

```diff
@@ -0,0 +1,184 @@
+package platform
+
+// Largely based on
```

Comment:
Then we should add `// Copyright 2013-2018 Docker, Inc.`

Reply:
(Still WIP)